assets/src/edit-story/components/library/panes/media/mediaElement.js

```diff
@@ -170,11 +171,22 @@ const MediaElement = ({
   const onClick = () => onInsert(resource, width, height);
 
   if (type === 'image') {
+    let imageSrc = src;
+    if (sizes) {
+      const { web_stories_thumbnail: webStoriesThumbnail, large, full } = sizes;
```

Comment:
What about checking the existing `thumbnail` size?

Reply:
Or perhaps even looping through the sizes and taking the first one that's at least `150` wide (or the full size if that doesn't exist)?

Reply:
Also wondering if it might make sense to filter the sizes in PHP and set the most suitable sizes already there. There might be setups with a lot of sizes :)

Thoughts?

Reply:
So thumbnail normally crops to 150. Which is why I didn't both with it. The reason I went with large and full as both are uncropped.

Reply:
Alright, makes sense, it's to ensure that cropped image wouldn't be used then.